### ISSUE TITLE
netip in Go 1.18

### ISSUE THREAD

**Issue by beriberikix** [plgd-dev/go-coap#313]
https://pkg.go.dev/net/netip

Go 1.18 introduces a new IP address format, that among other things, reduces memory usage ([background](https://tailscale.com/blog/netaddr-new-ip-type-for-go/).)

Are there plans to support 1.18 and adopt netip?

**Comment by HRogge** [comment]
The new netip package is definitely a HUGE step forward (we have been using it predecessor for quite a while)... unfortunately its slightly incompatible with its earlier revision (https://pkg.go.dev/inet.af/netaddr). So moving to netip would mean loosing compatibility with go 1.17 and earlier.

and some IDEs are still a bit shaky in terms of the new Go 1.18 features (generics!)

**Comment by beriberikix** [comment]
> So moving to netip would mean loosing compatibility with go 1.17 and earlier.

Yup, that's what I figured would be the limiting factor. Is there a specific commitment to versions or backwards compat?

**Comment by HRogge** [comment]
Minimum Go version of Go-Coap has been changed to 1.18... maybe its time to reinvestigate?

There are not only net.IP, we might also look into replacing net.Addr in some functions (its a really horrible interface to work with).
